sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/test/scala/com/azure/cosmos/spark/SparkE2EChangeFeedITest.scala

```diff
@@ -178,6 +187,159 @@ class SparkE2EChangeFeedITest
+
+    val sourceContainer = cosmosClient.getDatabase(cosmosDatabase).getContainer(cosmosContainer)
+
+    val testId = UUID.randomUUID().toString.replace("-", "")
+    val targetContainerResponse = cosmosClient.getDatabase(cosmosDatabase).createContainer(
+      "target_" + testId,
+      "/id",
+      ThroughputProperties.createManualThroughput(18000)).block()
+    val targetContainer = cosmosClient
+      .getDatabase(cosmosDatabase)
+      .getContainer(targetContainerResponse.getProperties.getId)
+
+    // Initially ingest 100 records
+    for (i <- 0 until 100) {
+      val objectNode = Utils.getSimpleObjectMapper.createObjectNode()
+      objectNode.put("name", "Shrodigner's cat")
+      objectNode.put("type", "cat")
+      objectNode.put("age", 20)
+      objectNode.put("sequenceNumber", i)
+      objectNode.put("id", UUID.randomUUID().toString)
+      sourceContainer.createItem(objectNode).block()
+    }
+
+    Thread.sleep(1000)
+
+    val changeFeedCfg = Map(
+      "spark.cosmos.accountEndpoint" -> cosmosEndpoint,
+      "spark.cosmos.accountKey" -> cosmosMasterKey,
+      "spark.cosmos.database" -> cosmosDatabase,
+      "spark.cosmos.container" -> cosmosContainer,
+      "spark.cosmos.read.inferSchemaEnabled" -> "false"
+    )
+
+    val writeCfg = Map(
+      "spark.cosmos.accountEndpoint" -> cosmosEndpoint,
+      "spark.cosmos.accountKey" -> cosmosMasterKey,
+      "spark.cosmos.database" -> cosmosDatabase,
+      "spark.cosmos.container" -> targetContainer.getId,
+      "spark.cosmos.write.strategy" -> "ItemOverwrite",
+      "spark.cosmos.write.bulkEnabled" -> "true",
+      "checkpointLocation" -> ("/tmp/" + testId + "/")
+    )
+
+    val processedRecordCount = new AtomicLong()
+    spark.streams.addListener(new StreamingQueryListener() {
+      override def onQueryStarted(queryStarted: QueryStartedEvent): Unit = {}
+      override def onQueryTerminated(queryTerminated: QueryTerminatedEvent): Unit = {}
+      override def onQueryProgress(queryProgress: QueryProgressEvent): Unit = {
+        processedRecordCount.addAndGet(queryProgress.progress.sink.numOutputRows)
+      }
+    })
+
+    val changeFeedDF = spark
+      .readStream
+      .format("cosmos.changeFeed")
+      .options(changeFeedCfg)
+      .load()
+
+    var microBatchQuery = changeFeedDF
+      .writeStream
+      .format("cosmos.items")
+      .queryName(testId)
+      .options(writeCfg)
+      .outputMode("append")
+      .start()
+
+    microBatchQuery.processAllAvailable()
+
+    Thread.sleep(1000)
+
+    var sourceCount: Long = getRecordCountOfContainer(sourceContainer)
+    logInfo(s"RecordCount in source container after first execution: $sourceCount")
+    var targetCount: Long = getRecordCountOfContainer(targetContainer)
+    logInfo(s"RecordCount in target container after first execution: $targetCount")
+
+    processedRecordCount.get() shouldEqual 100L
+    sourceCount shouldEqual 100L
+    sourceCount shouldEqual targetCount
+
+    spark.close()
```

Comment:
`SparkE2EChangeFeedITest` uses a pre-created spark which is not meant to be closed by the test.

as a pattern if you need to close or re-initialize spark, I would suggest you move the test to a new class which doesn't inherit `Spark` test trait

Reply:
Good idea - Thanks. Done